### DISCUSSION TITLE
How enable TLS for MySQL

### DISCUSSION THREAD

**Discussion by chillivilli** [apache/airflow#20293] [Q&A]
I am trying to set up TLS support for:
**sql_alchemy_conn
result_backend**

What option should I use in the connection string to tell Airflow to use TLS to connect to the database?

**Comment by potiuk** [comment]
for sql_alchemy_conn - Look at this and related sqlalchemy documentation https://airflow.apache.org/docs/apache-airflow/stable/configurations-ref.html#sql-alchemy-connect-args -

**Reply by potiuk** [reply]
You simply need to find out from sqlalchemy docs what should be your connect args

**Reply by potiuk** [reply]
For result backend see the parameters here (below result-backend you can see the ssl parameters): https://airflow.apache.org/docs/apache-airflow/stable/configurations-ref.html#result-backend

**Comment by chillivilli** [comment]
Thanks for your answers, but the documentation does not explicitly specify which option in the connection string to add, to activate TLS. MySQL client connects without any problem with TLS. Airflow, on the other hand, does not.

The source lines look like this:

sql_alchemy_conn = mysql+mysqldb://user:password@mysqlhost:3306/airflow 
result_backend = db+mysql://user:password@mysqlhost:3306/airflow

I tried adding ?ssl=true to the end of the line but it does not work. 
If anyone has been able to enable TLS for MySQL in Airflow, please let me know how you did it.

ssl_active = true also doesn`t work. And I would like to avoid using a client certificate.

**Reply by potiuk** [reply]
Airlfow uses sqlalchemy under the hood for DB connetion so you need to consult the sqlalchemy docs.

Did you have a look at the linked sqlalchemy documentation (you find the link next? 

I think it explains it pretty well and tells you what to do next.  

Note that you need to find documentation for the specific sqlalchemy driver. There is no single "common" way that will work - you can connect to mysql (and other dbs) using various drivers and each driver has it's own way of passing parameters and you have to find out which driver you use and which parameters it supports. All can be found in the sqlalchemy documentation of the specific driver.

**Reply by potiuk** [reply]
For example for PyMSQL driver you can find this documentaiton in SQLAlchemy https://docs.sqlalchemy.org/en/14/dialects/mysql.html#pymysql-ssl  - but this is just this specific PyMSQL driver, you might use a dfferent one with different parameters.

**Reply by chillivilli** [reply]
Thank you for your reply.
For MySQL, I see this construction in the documentation.
connection_uri = (
    "mysql+mysqldb://scott:tiger@192.168.0.134/test"
    "?ssl_ca=/home/gord/client-ssl/ca.pem"
    "&ssl_cert=/home/gord/client-ssl/client-cert.pem"
    "&ssl_key=/home/gord/client-ssl/client-key.pem"
)
It uses a client certificate. But I would not like to use a client certificate, I would like to have just TLS connection to the database. With certificate check based on system CA certificates. 
And unfortunately, I do not see such a possibility in the documentation.

**Reply by potiuk** [reply]
I think it's a question that you can ask in SQLAlchemy. Maybe they can help you. Feel free to report here the result of your inquiry, but I believe this is a deliberate decision on their side. Courious to see what you find though!
